src/app.js

```diff
@@ -1,56 +1,73 @@
-import React, { Component } from "react";
+import React, { useLayoutEffect, useEffect, useState } from "react";
 import PropTypes from "prop-types";
 import { Root, Routes } from "react-static";
 import { Route } from "react-router";
 import { ThemeProvider } from "styled-components";
+import { animateScroll as scroll } from "react-scroll";
+
 import GlobalStyle from "./styles/global";
 import theme from "./styles/theme";
 import Analytics from "./google-analytics";
 import NotFound from "./pages/404";
 
-const scrollContent = async ({ hash }, contentPaneClass = ".Page-content") => {
+const HEADER_PIXEL_HEIGHT = theme.layout.headerHeight.split("rem")[0] * 10;
+const SCROLL_PIXEL_OFFSET = 25;
+const DEFAULT_PAGE_CONTENT_CLASS = ".Page-content";
+const ROUTES = ["docs", "faq", "guides"];
+
+const scrollContent = async (
+  hash,
+  contentPaneClass = DEFAULT_PAGE_CONTENT_CLASS
+) => {
   const item = document.querySelector(`${contentPaneClass} ${hash}`);
-  if (item) {
-    item.scrollIntoView();
-  }
-};
 
-const scrollSidebar = async (location, activeItemClass = ".is-active") => {
-  const item = document.querySelector(activeItemClass);
-  if (item) {
-    item.scrollIntoView();
+  if (!item) {
+    return;
   }
+
+  const rect = item.getBoundingClientRect();
+  const truePosition =
+    (rect.top + window.pageYOffset || document.documentElement.scrollTop) -
+    HEADER_PIXEL_HEIGHT -
+    SCROLL_PIXEL_OFFSET;
+
+  scroll.scrollTo(truePosition, {
+    duration: 500,
+    delay: 0,
+    smooth: "easeOutQuad"
+  });
 };
 
-const checkScrollRoutes = (pathname, routes = ["docs", "faq", "guides"]) =>
+const checkScrollRoutes = (pathname, routes = ROUTES) =>
   routes.some(r => pathname.includes(r));
 
-class ScrollToTop extends Component {
-  componentDidMount() {
-    if (
-      typeof window !== "undefined" &&
-      checkScrollRoutes(this.props.location.pathname)
-    ) {
-      scrollContent(this.props.location);
-      scrollSidebar(this.props.location);
-    }
-  }
+const ScrollToCurrentSection = ({ location, children }) => {
+  const { pathname, hash = "" } = location;
+
+  const [pageContentHeight, setPageContentHeight] = useState();
+
+  const pageContentHeightObserver = new ResizeObserver(element => {
```

Comment:
Do we need to polyfill this for Edge and Safari? 
https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver

Reply:
yes you are absolutely right, i realized i'd forgotten this as soon as I got on the lightrail last night, adding one this AM. thank you for catching this!!!